autotx/utils/agent/build_goal.py

```diff
@@ -66,35 +66,61 @@ def build_goal(prompt: str, agents_information: str, smart_account_address: ETHA
 def analyze_user_prompt(chat_history: str, agents_information: str, smart_account_address: ETHAddress) -> DefineGoalResponse:
     template = dedent(
         """
-        Based on the following chat history between you and the user: 
-        ```
-        {chat_history}
-        ```
-            
+        You are a task assignment expert. You are good at analyzing the user's chat records to complete the task goals and conditions represented in the user's chat records.
         You must analyze the prompt and define a goal to be executed by the agents.
         If the prompt is not clear or missing information, you MUST ask for more information.
         If the prompt is invalid, unsupported or outside the scope of the agents, you MUST ask for a new prompt.
         Always ensure you have all the information needed to define the goal that can be executed without prior context.
+        When dealing with Ethereum transactions, assume the following is the user's address: {smart_account_address}
         
         The available agents and tools:
         {agents_information}
 
         Respond ONLY in one of three of the following JSON formats:
-        1:
+        
         {{
             "type": "goal",
-            "goal": "The detailed goal here. No need to mention specific agents or tools. But you MUST mention the user's address."
+            "goal": "The detailed goal here. No need to mention specific agents or tools."
         }}
-        2:
+        
         {{
             "type": "missing_info",
             "message": "The information that is missing here"
         }}
-        3:
+        
         {{
             "type": "unsupported",
             "message": "Reason why the prompt is unsupported here"
         }}
+        
+        Please note: You can only output content in json format, and do not output any other content!
```

Comment:
this line is repeated in line 121 - is that intended?

Reply:
Yes, we have found that for small parameter models, emphasizing the most important content before and after a long prompt seems to be helpful for the stability of the output.

Reply:
Or can we use different prompts for strong inference models such as GPT-4 or small parameter models that need to constantly emphasize requirements? It is true that long prompts may not be necessary for gpt-4, which can already handle it well.

Reply:
> Or can we use different prompts for strong inference models such as GPT-4 or small parameter models that need to constantly emphasize requirements? It is true that long prompts may not be necessary for gpt-4, which can already handle it well.

i think it make sense to have one prompt for gpt-4 and other for small parameter models

based on the benchmark results from https://github.com/polywrap/AutoTx/pull/119#pullrequestreview-1973166962 i would recommend updating the examples with the changes i suggested, see if they improve the planning (which they should, since the examples rn are incorrect) and if it does not improve, then use different prompts for the different models :smile:

Reply:
Yes for right now I think we should have a single prompt. Adding more will just add another dimension of complexity which will make it harder to get consistent results.

Reply:
Yes, I agree with you. In the case where gpt-4 can already handle the requirements of concise prompts, more examples may only consume tokens. I think two different sets of prompts can be submitted to adapt to gpt. -4 and our local llms